lib/src/logger/deprecation_processing.dart

```diff
@@ -45,7 +45,10 @@ final class DeprecationProcessingLogger extends LoggerWithDeprecationType {
       {required this.silenceDeprecations,
       required this.fatalDeprecations,
       required this.futureDeprecations,
-      this.limitRepetition = true}) {
+      this.limitRepetition = true});
+
+  /// Warns if any of the deprecations options are incompatible or unnecessary.
+  void validate() {
```

Comment:
Why split this out into a separate method?

Reply:
We only want to emit warnings for invalid/unnecessary deprecation flags once and we're now wrapping loggers more than once in many cases, so the warning part is separated out from the constructor and only called on the main wrapped logger.